# Análisis exploratorio 

* Luisa Merlo García | A01067715
* Edgar Osvaldo Navarro | A01644488
* Ariana Guadalupe Rosales Villalobos | A01644773

## Cargamos el Dataset y revisamos sus dimensiones

In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from pathlib import Path
from IPython.display import display

# Ajusta la carpeta si el Excel no está junto al notebook
archivo = Path("Datos/Resumen Downtime Current.xlsx")

df_original = pd.read_excel(archivo, sheet_name="2024")

# Trabajamos con una copia para conservar los datos originales
df = df_original.copy()
df.columns = df.columns.str.strip()

print("Forma del dataset:", df.shape)
print("Cantidad de registros:", df.shape[0])
print("Cantidad de variables:", df.shape[1])

display(df.head())
df.info()

print("\nFecha mínima de apertura:", df["Opened"].min())
print("Fecha máxima de apertura:", df["Opened"].max())

Forma del dataset: (10167, 28)
Cantidad de registros: 10167
Cantidad de variables: 28


,Equipo,Opened,Update,Reason,Ticket,Downtime,ISSUE 13,Description 9,Work Description (Close Details) 17,Dia,...,Año,Fecha completa,2nd class 29,ss,Corrective,Class2 29,Eq Type 26,Automotive? As on Aug-2024,Brand,Model
0,Preferida,2026-08-23 23:50:52.500,2026-08-24 00:14:31.837,MANTENIMIENTO - MANTTO. CORRECTIVO,REQ-1760551,0.39426,Mechanical Failure,Favor de checar giro intermedio de fin de braz...,Se extrae tonillo capado de sujecion de block ...,2026-08-23 23:50:52.500,...,2026,2026 - 8,Robot Wrist problem,1,Corrective,Robot Wrist problem,Robot,NO,SEPRO,S5-35 S3 AXIAL
1,LaBarranca,2026-08-23 23:25:43.213,2026-08-23 23:27:48.213,MANTENIMIENTO - MANTTO. CORRECTIVO,REQ-1760257,0.034722,Calibration Parameter Modification,Favor de revisar falla de presion de cierre,Restablece maquina 2 veces en turno\nSe ajusta...,2026-08-23 23:25:43.213,...,2026,2026 - 8,Clamping/ejector movement problems,1,Corrective,Clamping/ejector movement problems,IMM,NO,KRAUSS MAFFEI,KM420-1900C2
2,Ruisenores,2026-08-23 23:21:46.997,2026-08-24 00:05:28.297,MANTENIMIENTO - MANTTO. CORRECTIVO,REQ-1760550,0.728139,Clamping Piston Failure,Favor de checar nivel de aceite bajo se apaga ...,Realiza recuperacion de nivel de aceite en maq...,2026-08-23 23:21:46.997,...,2026,2026 - 8,Oil Alarm,1,Corrective,Oil Alarm,IMM,NO,NISSEI,FV9000
3,LaBarranca,2026-08-23 21:23:06.703,2026-08-23 21:28:53.060,MANTENIMIENTO - MANTTO. CORRECTIVO,REQ-1760257,0.09621,Calibration Parameter Modification,Favor de revisar falla de presion de cierre,Restablece maquina 2 veces en turno\nSe ajusta...,2026-08-23 21:23:06.703,...,2026,2026 - 8,Clamping/ejector movement problems,1,Corrective,Clamping/ejector movement problems,IMM,NO,KRAUSS MAFFEI,KM420-1900C2
4,Canada,2026-08-23 20:54:49.517,2026-08-23 21:05:09.210,MANTENIMIENTO - MANTTO. CORRECTIVO,REQ-1760499,0.172137,Process Adjustment,Favor de colocar barra donde se embobinado ro...,se volvio a fijar barra de el winder,2026-08-23 20:54:49.517,...,2026,2026 - 8,Foil feeder,1,Corrective,Foil feeder,Feeder,NO,WITTMANN,Feeder


<class 'pandas.DataFrame'>
RangeIndex: 10167 entries, 0 to 10166
Data columns (total 28 columns):
 #   Column                               Non-Null Count  Dtype         
---  ------                               --------------  -----         
 0   Equipo                               10167 non-null  str           
 1   Opened                               10167 non-null  datetime64[us]
 2   Update                               10167 non-null  datetime64[us]
 3   Reason                               10167 non-null  str           
 4   Ticket                               10162 non-null  str           
 5   Downtime                             10167 non-null  object        
 6   ISSUE 13                             8383 non-null   object        
 7   Description 9                        8386 non-null   object        
 8   Work Description (Close Details) 17  8384 non-null   object        
 9   Dia                                  10167 non-null  datetime64[us]
 10  mes                  

El dataset contiene 10,167 registros y 28 variables. Cada fila representa un registro de downtime asociado a un equipo y, cuando está disponible, a un ticket. Un mismo ticket puede aparecer en varios registros, por lo que el número de filas no equivale al número de tickets únicos. Aunque la hoja se llama “2024”, las fechas de apertura abarcan del 2 de enero de 2025 al 23 de agosto de 2026.

## Identificamos los tipos de variables


In [11]:
tipos_variables = {
    "Equipo": "Categórica nominal",
    "Opened": "Temporal: fecha y hora",
    "Update": "Temporal: fecha y hora",
    "Reason": "Categórica nominal",
    "Ticket": "Identificador",
    "Downtime": "Cuantitativa continua",
    "ISSUE 13": "Categórica nominal",
    "Description 9": "Texto libre",
    "Work Description (Close Details) 17": "Texto libre",
    "Dia": "Temporal: fecha y hora",
    "mes": "Temporal: mes",
    "Semana": "Temporal: año y semana",
    "Equipo 26": "Categórica nominal",
    "Sub system 27": "Categórica nominal",
    "EQ2": "Identificador compuesto de equipo",
    "Clase 13": "Categórica nominal",
    "DD": "Identificador compuesto",
    "Device 28": "Categórica nominal",
    "Año": "Temporal: año",
    "Fecha completa": "Temporal: año y mes",
    "2nd class 29": "Categórica nominal",
    "ss": "Código numérico por documentar",
    "Corrective": "Categórica nominal",
    "Class2 29": "Categórica nominal",
    "Eq Type 26": "Categórica nominal",
    "Automotive? As on Aug-2024": "Categórica nominal",
    "Brand": "Categórica nominal",
    "Model": "Categórica nominal"
}

resumen_variables = pd.DataFrame({
    "Tipo en pandas": df.dtypes.astype(str),
    "Tipo de variable": pd.Series(tipos_variables),
    "Valores únicos": df.nunique()
})

display(resumen_variables)

,Tipo en pandas,Tipo de variable,Valores únicos
Equipo,str,Categórica nominal,92
Opened,datetime64[us],Temporal: fecha y hora,10159
Update,datetime64[us],Temporal: fecha y hora,10145
Reason,str,Categórica nominal,4
Ticket,str,Identificador,5000
Downtime,object,Cuantitativa continua,10143
ISSUE 13,object,Categórica nominal,95
Description 9,object,Texto libre,4594
Work Description (Close Details) 17,object,Texto libre,3873
Dia,datetime64[us],Temporal: fecha y hora,10159


Observamos que predominan las variables categricas,los identificadores y campos temporales. Downtime corresponde a una metrica de duración pero pandas lo carga como object.

## Revisar valores faltantes

In [12]:
# Contamos valores faltantes de cada columna
faltantes = pd.DataFrame({
    "Cantidad": df.isna().sum(),
    "Porcentaje": (df.isna().mean() * 100).round(2)
})

faltantes = faltantes.sort_values("Cantidad", ascending=False)

display(faltantes)

print("Total de celdas faltantes:", df.isna().sum().sum())
print("Filas con al menos un faltante:", df.isna().any(axis=1).sum())

# También buscamos texto vacio o  espacios
columnas_texto = df.select_dtypes(include=["object", "string"]).columns

vacios = {}

for columna in columnas_texto:
    valores = df[columna].astype("string")
    vacios[columna] = valores.str.strip().eq("").sum()

display(
    pd.Series(vacios, name="Textos vacíos")
    .sort_values(ascending=False)
)

,Cantidad,Porcentaje
ISSUE 13,1784,17.55
Clase 13,1784,17.55
Sub system 27,1783,17.54
Work Description (Close Details) 17,1783,17.54
Eq Type 26,1783,17.54
Description 9,1781,17.52
Model,1605,15.79
Brand,1491,14.67
Class2 29,1354,13.32
2nd class 29,1349,13.27


Total de celdas faltantes: 20889
Filas con al menos un faltante: 2479


Equipo                                 0
Reason                                 0
Ticket                                 0
Downtime                               0
ISSUE 13                               0
Description 9                          0
Work Description (Close Details) 17    0
Semana                                 0
Equipo 26                              0
Sub system 27                          0
EQ2                                    0
Clase 13                               0
DD                                     0
Device 28                              0
Fecha completa                         0
2nd class 29                           0
Corrective                             0
Class2 29                              0
Eq Type 26                             0
Automotive? As on Aug-2024             0
Brand                                  0
Model                                  0
Name: Textos vacíos, dtype: int64

Entre las variables con más faltantes están ISSUE 13 y Clase 13, con 1,784 cada una, y los campos de descripción y clasificación del equipo. Ticket tiene cinco valores faltantes. No se eliminaran estas filas ya que existen campos con informacion util en otros casos.

In [13]:
# Revisamos posibles informacion faltante, marcada como 0
reporte_marcadores = []

for columna in columnas_texto:
    valores = df[columna].astype("string").str.strip().str.casefold()

    reporte_marcadores.append({
        "Variable": columna,
        "Ceros escritos o numéricos": valores.isin(["0", "0.0"]).sum(),
        "Contiene 'sin ticket'": valores.str.contains(
            "sin ticket", regex=False, na=False
        ).sum()
    })

reporte_marcadores = pd.DataFrame(reporte_marcadores)

display(
    reporte_marcadores[
        (reporte_marcadores["Ceros escritos o numéricos"] > 0) |
        (reporte_marcadores["Contiene 'sin ticket'"] > 0)
    ]
)

,Variable,Ceros escritos o numéricos,Contiene 'sin ticket'
2,Ticket,0,1765
3,Downtime,0,1
4,ISSUE 13,1602,0
6,Work Description (Close Details) 17,1507,0
8,Equipo 26,154,0
9,Sub system 27,3281,0
11,Clase 13,1602,0
18,Eq Type 26,154,0


Se observan ceros en campos descriptivos y de clasificación. Estos podrían representar información no registrada o informacion que no es necesari a en ese caso. Además, 1,765 registros contienen “Sin Ticket”, incluyendo variantes y descripciones adicionales.

## Revisar duplicados 

In [14]:
# Duplicados considerando todas las columnas
duplicados_completos = df.duplicated()

print("Filas completamente duplicadas:", duplicados_completos.sum())

# Posibles registros repetidos del mismo evento
clave_evento = ["Ticket", "Equipo", "Opened", "Update"]

print(
    "Repeticiones de la clave del evento:",
    df.duplicated(subset=clave_evento).sum()
)

# Revisamos la repetición de identificadores con formato REQ-número
tickets = df["Ticket"].astype("string").str.strip()
es_req = tickets.str.fullmatch(r"REQ-\d+", na=False)

frecuencia_tickets = tickets[es_req].value_counts()

print("\nRegistros con formato REQ-número:", es_req.sum())
print("Tickets REQ distintos:", tickets[es_req].nunique())
print(
    "Apariciones adicionales de tickets REQ:",
    tickets[es_req].duplicated().sum()
)

display(
    frecuencia_tickets[frecuencia_tickets > 1]
    .head(10)
    .rename("Cantidad de registros")
)

# Mostramos un ejemplo del ticket más frecuente
if not frecuencia_tickets.empty:
    ticket_ejemplo = frecuencia_tickets.index[0]

    display(
        df.loc[
            tickets.eq(ticket_ejemplo).fillna(False),
            ["Ticket", "Equipo", "Opened", "Update", "Downtime"]
        ].sort_values("Opened").head(10)
    )

Filas completamente duplicadas: 0
Repeticiones de la clave del evento: 0

Registros con formato REQ-número: 8388
Tickets REQ distintos: 4931
Apariciones adicionales de tickets REQ: 3457


Ticket
REQ-1684425    102
REQ-1518418     54
REQ-1682059     32
REQ-1678991     28
REQ-1721473     25
REQ-1602471     22
REQ-1742067     20
REQ-1760257     19
REQ-1730625     17
REQ-1603477     17
Name: Cantidad de registros, dtype: Int64

,Ticket,Equipo,Opened,Update,Downtime
2389,REQ-1684425,LaBarranca,2026-04-08 01:43:44.420,2026-04-08 01:47:27.543,0.061979
2388,REQ-1684425,LaBarranca,2026-04-08 02:13:47.460,2026-04-08 02:15:41.050,0.031553
2387,REQ-1684425,LaBarranca,2026-04-08 03:08:44.540,2026-04-08 03:10:06.147,0.022669
2386,REQ-1684425,LaBarranca,2026-04-08 04:13:32.713,2026-04-08 04:15:28.520,0.032169
2384,REQ-1684425,LaBarranca,2026-04-08 06:31:59.750,2026-04-08 06:33:47.760,0.030003
2380,REQ-1684425,LaBarranca,2026-04-08 11:21:34.440,2026-04-08 11:22:13.137,0.010749
2378,REQ-1684425,LaBarranca,2026-04-08 12:27:27.073,2026-04-08 12:28:25.107,0.016121
2376,REQ-1684425,LaBarranca,2026-04-08 13:47:04.073,2026-04-08 13:51:14.053,0.069439
2374,REQ-1684425,LaBarranca,2026-04-08 16:21:49.817,2026-04-08 16:25:20.510,0.058526
2372,REQ-1684425,LaBarranca,2026-04-08 16:55:07.113,2026-04-08 16:59:44.310,0.076999


No se encontraron filas completamente duplicadas ni repeticiones de la combinación Ticket, Equipo, Opened y Update. dentificaron 8,388 registros con formato REQ-número, correspondientes a 4,931 tickets distintos. Las 3,457 apariciones adicionales no deben eliminarse,aún ya que pueden representar diferentes eventos asociados al mismo ticket.

## Revisamos valores inconsistentes 

In [15]:
# Convertimos en variables auxiliares para detectar errores
downtime = pd.to_numeric(df["Downtime"], errors="coerce")
opened = pd.to_datetime(df["Opened"], errors="coerce")
update = pd.to_datetime(df["Update"], errors="coerce")

# Valores presentes que no pudieron convertirse
downtime_no_numerico = df["Downtime"].notna() & downtime.isna()
opened_invalido = df["Opened"].notna() & opened.isna()
update_invalido = df["Update"].notna() & update.isna()

# Calculamos el intervalo en horas
duracion_calculada = (update - opened).dt.total_seconds() / 3600

# Comparamos únicamente los casos numéricos finitos y con ambas fechas
comparables = (
    downtime.notna() &
    np.isfinite(downtime) &
    opened.notna() &
    update.notna()
)

# Tolerancia de un segundo por posibles diferencias de precisión
diferencia_duracion = (
    comparables &
    ((downtime - duracion_calculada).abs() > 1 / 3600)
)

revision = pd.Series({
    "Downtime no numérico": downtime_no_numerico.sum(),
    "Downtime negativo": downtime.lt(0).sum(),
    "Downtime igual a cero": downtime.eq(0).sum(),
    "Downtime infinito": np.isinf(downtime).sum(),
    "Opened no convertible": opened_invalido.sum(),
    "Update no convertible": update_invalido.sum(),
    "Update anterior a Opened": (update < opened).sum(),
    "Downtime diferente del intervalo calculado": diferencia_duracion.sum(),
    "Mes fuera de 1 a 12": (~df["mes"].between(1, 12)).sum(),
    "Mes distinto al de Opened": (df["mes"] != opened.dt.month).sum(),
    "Año distinto al de Opened": (df["Año"] != opened.dt.year).sum()
}, name="Cantidad")

display(revision)

# Mostramos los registros que requieren revisión
casos_revision = df.loc[
    downtime_no_numerico | diferencia_duracion,
    ["Ticket", "Equipo", "Opened", "Update", "Downtime"]
].copy()

casos_revision["Duración calculada (horas)"] = duracion_calculada.loc[
    casos_revision.index
]

display(casos_revision)
display(downtime.describe())

Downtime no numérico                          1
Downtime negativo                             0
Downtime igual a cero                         0
Downtime infinito                             0
Opened no convertible                         0
Update no convertible                         0
Update anterior a Opened                      0
Downtime diferente del intervalo calculado    1
Mes fuera de 1 a 12                           0
Mes distinto al de Opened                     0
Año distinto al de Opened                     0
Name: Cantidad, dtype: int64

,Ticket,Equipo,Opened,Update,Downtime,Duración calculada (horas)
4946,Sin Ticket,Senai,2025-10-25 02:47:00.310,2025-10-25 02:55:29.707,Sin Ticket,0.141499
5999,REQ-1592932,Chamaco,2025-08-21 19:31:27.957,2025-08-21 20:19:51.460,0.016811,0.806529


count    10166.000000
mean         1.009662
std          2.210159
min          0.000084
25%          0.109107
50%          0.393640
75%          1.051657
max         67.824991
Name: Downtime, dtype: float64

Se encontró un registro con “Sin Ticket” dentro de Downtime, donde se esperaba un número. Tampoco se encontraron fechas Update anteriores a Opened, ni inconsistencias de mes o año respecto a la apertura. Se detectó un registro numérico cuya duración difiere del intervalo entre fechas en más de un segundo.

In [16]:

# REvisamos variables categoricas

espacios = {}

for columna in columnas_texto:
    valores = df[columna].astype("string")
    espacios[columna] = valores.ne(valores.str.strip()).sum()

espacios = pd.Series(espacios, name="Registros con espacios externos")

display(espacios[espacios > 0].sort_values(ascending=False))

# Frecuencias de campos que conviene revisar
for columna in ["Corrective", "Equipo 26", "Eq Type 26", "ss"]:
    print(f"\nValores de {columna}:")
    display(df[columna].value_counts(dropna=False))

# Categorías inesperadas según el significado aparente de Corrective
corrective = df["Corrective"].astype("string").str.strip()

categoria_inesperada = (
    corrective.notna() &
    ~corrective.isin(["Corrective", "NOT CORRECTIVE"])
)

display(
    df.loc[
        categoria_inesperada,
        ["Ticket", "Corrective", "2nd class 29"]
    ]
)


Description 9                          4705
Work Description (Close Details) 17     770
ISSUE 13                                338
Clase 13                                338
Model                                    31
Equipo 26                                 8
Eq Type 26                                8
Name: Registros con espacios externos, dtype: int64


Valores de Corrective:


Corrective
Corrective                   7716
NaN                          1258
NOT CORRECTIVE               1192
Failure in axis movements       1
Name: count, dtype: int64


Valores de Equipo 26:


Equipo 26
IMM                   5056
Robot                 2050
NaN                    963
Secador                458
Termoregulador         456
Blender                345
Cargador               224
0                      154
Fixture                142
Controlador            100
Feeder                  61
Security Cage           42
Horno UV                32
Cortadora               16
Hydraulic Unit          10
Foil Feeder              8
Slider                   8
hs                       7
Unidad Hidraulica        6
Foil Feeder              3
Dryer                    3
robot                    3
controlador              2
HS                       2
Hot Pack                 2
Molde                    2
Jalisco                  2
Andom                    2
Spare parts wh           2
Control Hot Runner       1
Hot pack                 1
Coradora                 1
fixture                  1
Power unit               1
hot Pack                 1
Name: count, dtype: int64


Valores de Eq Type 26:


Eq Type 26
IMM                   4279
Robot                 2035
NaN                   1783
Secador                452
Termoregulador         449
Blender                345
Cargador               223
0                      154
Fixture                136
Controlador            100
Feeder                  61
Security Cage           42
Horno UV                32
Cortadora               16
Foil Feeder              8
Slider                   8
hs                       7
Hydraulic Unit           7
Unidad Hidraulica        6
Foil Feeder              3
Dryer                    3
robot                    3
controlador              2
Molde                    2
Jalisco                  2
Andom                    2
Hot Pack                 1
Control Hot Runner       1
Hot pack                 1
Coradora                 1
fixture                  1
Power unit               1
hot Pack                 1
Name: count, dtype: int64


Valores de ss:


ss
1    10161
2        6
Name: count, dtype: int64

,Ticket,Corrective,2nd class 29
3758,Sin Ticket - Robot,Failure in axis movements,Failure in axis movements


Se encontraron espacios externos en siete columnas. La más afectada es Description 9, con 4,705 registros, seguida de Work Description (Close Details) 17, con 770. Por lo que se manejará con un .split() para que no se distingan un mismo valor no se considere como valores distintos.

Tambien econtramos un posible error de captura en Failiure axis movement.

Tambien se observamn inconsistencias en algunos vaalores que existen en las identidades de Equipo 26 y ED Type 26. Con diferencias de mayusculas que generan categorias separadas.

## Revisamos variables redundantes:



In [17]:
pares = [
    ("Opened", "Dia"),
    ("ISSUE 13", "Clase 13"),
    ("2nd class 29", "Class2 29"),
    ("Equipo 26", "Eq Type 26")
]

for columna_1, columna_2 in pares:
    iguales = (
        df[columna_1].eq(df[columna_2]) |
        (df[columna_1].isna() & df[columna_2].isna())
    )

    print(
        f"{columna_1} / {columna_2}: "
        f"{(~iguales).sum()} registros diferentes"
    )

print("\nColumnas constantes:")
display(df.nunique(dropna=False)[df.nunique(dropna=False) == 1])

Opened / Dia: 0 registros diferentes
ISSUE 13 / Clase 13: 0 registros diferentes
2nd class 29 / Class2 29: 8 registros diferentes
Equipo 26 / Eq Type 26: 820 registros diferentes

Columnas constantes:


Automotive? As on Aug-2024    1
dtype: int64

Opened y Dia son idénticas, al igual que ISSUE 13 y Clase 13. Sin embargo, 2nd class 29 y Class2 29 difieren en ocho registros, mientras que Equipo 26 y Eq Type 26 difieren en 820. Por ello, estas últimas no deben tratarse como equivalentes sin revisión. Tambien se encontró Automotive? As on Aug-2024 tiene únicamente el valor NO.

## Aplicamos limpieza de espacios y variables mayusculas

In [19]:
# Eliminamos columnas redundantes o sin variación
columnas_eliminar = [
    "Dia",
    "mes",
    "Año",
    "Fecha completa",
    "Semana",
    "Clase 13",
    "DD",
    "Automotive? As on Aug-2024"
]

# Creamos una tabla nueva para conservar df completo
df_eda = df.drop(columns=columnas_eliminar).copy()

print("Dimensiones originales:", df.shape)
print("Dimensiones después de eliminar columnas:", df_eda.shape)

display(df_eda.head())

Dimensiones originales: (10167, 28)
Dimensiones después de eliminar columnas: (10167, 20)


,Equipo,Opened,Update,Reason,Ticket,Downtime,ISSUE 13,Description 9,Work Description (Close Details) 17,Equipo 26,Sub system 27,EQ2,Device 28,2nd class 29,ss,Corrective,Class2 29,Eq Type 26,Brand,Model
0,Preferida,2026-08-23 23:50:52.500,2026-08-24 00:14:31.837,MANTENIMIENTO - MANTTO. CORRECTIVO,REQ-1760551,0.39426,Mechanical Failure,Favor de checar giro intermedio de fin de braz...,Se extrae tonillo capado de sujecion de block ...,Robot,0,Robot - Preferida,C axis screw,Robot Wrist problem,1,Corrective,Robot Wrist problem,Robot,SEPRO,S5-35 S3 AXIAL
1,LaBarranca,2026-08-23 23:25:43.213,2026-08-23 23:27:48.213,MANTENIMIENTO - MANTTO. CORRECTIVO,REQ-1760257,0.034722,Calibration Parameter Modification,Favor de revisar falla de presion de cierre,Restablece maquina 2 veces en turno\nSe ajusta...,IMM,0,IMM - LaBarranca,Calibration parameter,Clamping/ejector movement problems,1,Corrective,Clamping/ejector movement problems,IMM,KRAUSS MAFFEI,KM420-1900C2
2,Ruisenores,2026-08-23 23:21:46.997,2026-08-24 00:05:28.297,MANTENIMIENTO - MANTTO. CORRECTIVO,REQ-1760550,0.728139,Clamping Piston Failure,Favor de checar nivel de aceite bajo se apaga ...,Realiza recuperacion de nivel de aceite en maq...,IMM,0,IMM - Ruisenores,Low oil level,Oil Alarm,1,Corrective,Oil Alarm,IMM,NISSEI,FV9000
3,LaBarranca,2026-08-23 21:23:06.703,2026-08-23 21:28:53.060,MANTENIMIENTO - MANTTO. CORRECTIVO,REQ-1760257,0.09621,Calibration Parameter Modification,Favor de revisar falla de presion de cierre,Restablece maquina 2 veces en turno\nSe ajusta...,IMM,0,IMM - LaBarranca,Calibration parameter,Clamping/ejector movement problems,1,Corrective,Clamping/ejector movement problems,IMM,KRAUSS MAFFEI,KM420-1900C2
4,Canada,2026-08-23 20:54:49.517,2026-08-23 21:05:09.210,MANTENIMIENTO - MANTTO. CORRECTIVO,REQ-1760499,0.172137,Process Adjustment,Favor de colocar barra donde se embobinado ro...,se volvio a fijar barra de el winder,Feeder,0,Feeder - Canada,Winder,Foil feeder,1,Corrective,Foil feeder,Feeder,WITTMANN,Feeder


In [21]:
# Partimos de la tabla con las ocho columnas eliminadas
df_limpio = df_eda.copy()

# Quitamos espacios externos solo a valores que sean texto.
# Conservamos los faltantes y los valores numéricos.
columnas_texto = df_limpio.select_dtypes(
    include=["object", "string"]
).columns

for columna in columnas_texto:
    df_limpio[columna] = df_limpio[columna].apply(
        lambda valor: valor.strip() if isinstance(valor, str) else valor
    )

# Unificamos variantes que solo cambian en mayúsculas
equivalencias = {
    "robot": "Robot",
    "fixture": "Fixture",
    "controlador": "Controlador",
    "hot pack": "Hot Pack",
    "hs": "HS"
}

for columna in ["Equipo 26", "Eq Type 26"]:
    df_limpio[columna] = df_limpio[columna].apply(
        lambda valor: equivalencias.get(valor.casefold(), valor)
        if isinstance(valor, str) else valor
    )

# Revisamos cómo quedaron las categorías
for columna in ["Equipo 26", "Eq Type 26"]:
    print(f"\nCategorías de {columna}:")
    display(df_limpio[columna].value_counts(dropna=False))


Categorías de Equipo 26:


Equipo 26
IMM                   5056
Robot                 2053
NaN                    963
Secador                458
Termoregulador         456
Blender                345
Cargador               224
0                      154
Fixture                143
Controlador            102
Feeder                  61
Security Cage           42
Horno UV                32
Cortadora               16
Foil Feeder             11
Hydraulic Unit          10
HS                       9
Slider                   8
Unidad Hidraulica        6
Hot Pack                 4
Dryer                    3
Molde                    2
Jalisco                  2
Andom                    2
Spare parts wh           2
Control Hot Runner       1
Coradora                 1
Power unit               1
Name: count, dtype: int64


Categorías de Eq Type 26:


Eq Type 26
IMM                   4279
Robot                 2038
NaN                   1783
Secador                452
Termoregulador         449
Blender                345
Cargador               223
0                      154
Fixture                137
Controlador            102
Feeder                  61
Security Cage           42
Horno UV                32
Cortadora               16
Foil Feeder             11
Slider                   8
HS                       7
Hydraulic Unit           7
Unidad Hidraulica        6
Hot Pack                 3
Dryer                    3
Molde                    2
Jalisco                  2
Andom                    2
Control Hot Runner       1
Coradora                 1
Power unit               1
Name: count, dtype: int64

Con esto podemos observar el número de cantidad de datos disponibles despúes de agrupar ñas etiquetas equivaneltentes sin eliminar filas y además se eliminaron las identidades de Dia, mes, Año, Fecha completa, Semana, Clase 13, DD y Automotive? As on Aug-2024 al ser variables redudnantes 

## Problemas de calidad identificados

El análisis exploratorio identificó problemas de calidad relacionados principalmente con valores faltantes, variables completamente vacías, categorías representadas mediante el valor "0", nombres de hojas que no corresponden con las fechas reales, categorías escritas de distintas maneras y tickets repetidos. No se encontraron filas completamente duplicadas ni tiempos negativos. Las fechas finales tampoco aparecen antes de las fechas iniciales.

Se recomienda eliminar únicamente las columnas completamente vacías o constantes que no tengan valor documental, normalizar las variables de texto, convertir los valores "0" a faltantes cuando correspondan, validar los tickets repetidos antes de eliminarlos y conservar los valores extremos de downtime hasta confirmar si representan eventos reales. Además, se necesita un diccionario de datos para interpretar variables como Valid, Val, ss, DD y las distintas columnas de clasificación.